### PR TITLE
feat: adding demo for dashed edges

### PR DESCRIPTION
**Pull request by sachaudh**
**What**: Closes #

**Additional issues**:

### CODE REVIEW

File: packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  SELECTION_EVENT,
+  TopologySideBar,
+  TopologyView,
+  useVisualizationController,
+  Visualization,
+  VisualizationProvider,
+  VisualizationSurface
+} from '@patternfly/react-topology';
+
+import './TopologyDemo.css';
```
Comment (alwayshooin) [inline packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx:14]: omg is this why it wasn't working for me 😭

File: packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          label: 'Node 0',
+          type: 'node',
+          width: 75,
+          height: 75,
+          data: {
+            id: 'node-0',
+            label: 'Node 0',
+            type: 'node',
+            width: 75,
+            height: 75
```
Comment (alwayshooin) [inline packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx:44]: i don't think you need to populate the `data` field for the node data model btw

File: packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      ]
+    };
+
+    function onSelect(ids: string[]) {
+      const newSelectedId = ids?.[0] || null;
+      setSelectedId(newSelectedId);
+    }
+
+    controller.fromModel(model, false);
+    controller.addEventListener(SELECTION_EVENT, onSelect);
```
Comment (alwayshooin) [inline packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx:86]: could we remove the side bar stuff for this demo?

File: packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx (hunk 2 above)
Comment (sachaudh) [reply packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx]: Oh yea I guess that is true. let me change that

File: packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx (hunk 3 above)
Comment (sachaudh) [reply packages/react-integration/demo-app-ts/src/components/demos/TopologyDemo/TopologyDashedEdgesDemo.tsx]: Yep let me do that too

**Pull request merged**